### PR TITLE
Don't hardcode LLVM version number

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

**Comment by gbaraldi** [comment]
Why don't we version LLVM on macos here?

File: base/binaryplatforms.jl [hunk 1]
```diff
@@ -904,7 +904,7 @@ function detect_cxxstring_abi()
     end
 
     function open_libllvm(f::Function)
-        for lib_name in ("libLLVM-14jl", "libLLVM", "LLVM", "libLLVMSupport")
+        for lib_name in (libllvm_name, "libLLVM", "LLVM", "libLLVMSupport")
```
Comment (giordano) [inline base/binaryplatforms.jl:907]: Should it be only libllvm_name?

Comment (vchuravy) [reply base/binaryplatforms.jl:907]: I will forward that question to @staticfloat

**Comment by vchuravy** [comment]
> Why don't we version LLVM on macos here?

I suspect that `@rpath` works better/just fine?

File: base/binaryplatforms.jl (hunk 1 above)
Comment (staticfloat) [reply base/binaryplatforms.jl:907]: Yeah, I think we've now successfully managed to abstract away the platform differences here.

File: base/Makefile [hunk 2]
```diff
@@ -57,6 +57,7 @@ else
 	@echo "const USE_GPL_LIBS = false" >> $@
 endif
 	@echo "const libllvm_version_string = \"$$($(LLVM_CONFIG_HOST) --version)\"" >> $@
+	@echo "const libllvm_name = \"$(LLVM_SHARED_LIB_NAME)\"" >> $@
```
Comment (inkydragon) [inline base/Makefile:60]: ci error:
```
Error in testset libLLVM_jll:
Error During Test at V:\julia\test\testdefs.jl:21
  Got exception outside of a @test
  LoadError: InitError: could not load library ".dll"
  The specified module could not be found.
```

local test
```sh
$ make -C base/ print-LLVM_SHARED_LIB_NAME
LLVM_SHARED_LIB_NAME=

$ make print-LLVM_SHARED_LIB_NAME
LLVM_SHARED_LIB_NAME=libLLVM-14jl
```

Perhaps this variable needs to be imported first.
https://github.com/JuliaLang/julia/blob/0a9abc1919a69b7b4d668e3f841b53b186448e41/Makefile#L3-L4

Comment (vchuravy) [reply base/Makefile:63]: Thanks!

**Pull request merged**